agent/tests/test_skills_executor.py

```
@@ -63,24 +63,54 @@ def test_unknown_tool_returns_error(self):
         # ready should be a bool
         assert isinstance(result["ready"], bool)
 
+    def test_not_ready_skips_operational_fields(self):
+        """When setup is not ready, operational fields should be absent."""
+        result = execute_tool("setup_and_operational_status", {})
+        if not result["ready"]:
+            assert "ai_provider" not in result
+            assert "ai_operational" not in result
+
+    @patch("iconfucius.health.urlopen")
+    def test_ready_includes_operational_fields(self, mock_urlopen, odin_project, monkeypatch):
```

Ruff (0.15.2)

[warning] Unused method argument: `odin_project`

(ARG002)